feat(bridge): display account proxy

shoom3301 opened the pull request:
# Summary

<<if there's an issue>>Fixes #issueNumber

High-level description of what your changes are accomplishing

Add screenshots if applicable. Images are nice :)

# To Test

1. <<Step one>> Open the page `about`

- [ ] <<What to expect?>> Verify it contains about information...
- [ ] Checkbox Style list of things a QA person could verify, i.e.
- [ ] Should display Text Input our storybook
- [ ] Input should not accept Numbers

2. <<Step two>> ...

# Background

Optional: Give background information for changes you've made, that might be difficult to explain via comments

elena-zh commented:
Hey @shoom3301 , cool, great job!
I'd like to propose some improvements, and report some issues.. :)
1. WDYT about moving 'Proxy account' to the 'Account' section? 
![move page to account](https://github.com/user-attachments/assets/3e28d456-c365-4a3e-bcff-4618de0478fc)
- It looks OK to a not connected user, so it an be always displayed there 
![image](https://github.com/user-attachments/assets/f28bee52-cf2d-4029-8481-61673a21f4be)

- Besides, it wil be possible to display it as a page, not like a modal, and you will not need to fix this UI case with shifted arrow to another line on a default view 
![image](https://github.com/user-attachments/assets/8e8e2f76-c663-45f4-af34-bb17c27f60c7)

- also, there will no need to display in in the Account modal
![no need](https://github.com/user-attachments/assets/20c14d83-ea3b-4e17-85d8-cdea73aee6ff)

- And there will no need to show this arrow **that is not clickable when open the page in another tab** (from order confirm modal, for example)
![image](https://github.com/user-attachments/assets/eeb51828-a3c5-456c-b565-50c62e213b22)

2. WDYT about separating FAQ between tabs? E.g. About section shows 'What is' quesiton, 'Recover funds' section shows an eponymous question? 
![separate questions per tabs](https://github.com/user-attachments/assets/11ac355e-5b5e-493c-9066-8a52a80587f2)

3. WDYT about removing 'fav tokens' section from the Recover funds token list? For me it looks a bit useless there, as I'm interested in my stuck funds there only
![fav tokens](https://github.com/user-attachments/assets/1f4a5545-49b3-4a9d-b088-ba0f87a31add)

4. WDYT about navigating a user directly to the 'Recover funds' tab from the Hooks link? 
![recover to the ep tab](https://github.com/user-attachments/assets/b229dfbd-1400-4753-8528-4d6c9b7042c7)


**Now, bugs:** 
1. [It might be not yet implemented] I created an order to another recipient, but it shows only Account proxy as a receivet. Might it be possible to separate these fields?  
![another recipient](https://github.com/user-attachments/assets/5dd71967-d538-44f0-b9b5-5fafa81546c9)
2. If I have >1 token on my proxy account, the warning shows the last one. IMO, it would be nice to have a bit more generic message saying that I have tokens on my proxy, and I can recover them from another tab
![message](https://github.com/user-attachments/assets/ec8413f0-f100-4b2c-a47a-2c4bb97c035d)
3. Should we still say to a user to check their stuck funds on blockexplorer if now we fetch them? 
![need step](https://github.com/user-attachments/assets/a3cc8291-1d74-4440-a749-ac7116bdfa83)

shoom3301 commented on apps/cowswap-frontend/src/modules/cowShed/containers/ProxyRecipient/index.tsx:
```diff
@@ -0,0 +1,84 @@
+
+  const { proxyAddress, account } = proxyAndAccount
+
+  if (recipient.toLowerCase() !== proxyAddress.toLowerCase()) {
+    throw new Error(
+      `Provided proxy address does not match CoW Shed proxy address!, recipient=${recipient}, proxyAddress=${proxyAddress}`,
+    )
+  }
```
It is intentional. If the provided values doesn't match proxy address we should not proceed

shoom3301 commented:
@elena-zh thanks for the feeback!

1. Added "Account proxy" item into header menu
2. Added "Connect wallet" button to the recover funds tab
3. Fixed account address displaying
4. I've left "Account proxy" link in the ACcount modal, I think it won't harm if we have it here. Also, it's only displayed if the proxy account is deployed (you've ever used cow hooks)
5. Removed "Favorite tokens" from the widget
6. Fixed back navigation from the page. Now if you came from Hooks page it will send you back to Hooks, otherwise - to Swap
7. WDYT about displaying a link to this page on Explorer? - not in this PR

After User research session we decided and I've implemented:
1. The Account proxy page should open as modal if you open it from Recipient tooltip
2. Added a green banner to confirm modal (the text is from Mindy)
3. Display proxy address instead of "Account proxy" in the recipient because

Bugs:
1. It will be changed in https://github.com/cowprotocol/cowswap/pull/5855
2. Added all tokens with balances displaying, see the pic bellow
3. Removed the point

<img width="510" alt="image" src="https://github.com/user-attachments/assets/5a71ecad-b903-418d-b0a0-a9f20f831728" />

shoom3301 commented:
@elena-zh another point - I've removed fast quote fetching in bridging mode. It doesn't makes sense, but it increases the load on bridging provider API (we face 429 errors more often)

elena-zh commented:
Hey @shoom3301 , thank you. 
However, still there are some questions and issues: 

1. 0 appears instead of a banner when data is being loaded: 
![image](https://github.com/user-attachments/assets/744b7ede-7df3-45eb-a002-1ee04c9ac102)
2. Could you please improve the address displaying on the Confirm modal and not to show an arrow in another line? 
![address a bit nicer](https://github.com/user-attachments/assets/8c0a4718-9413-4634-b05e-3552505de836)
3. Account Proxy address is displayed for an account that does not have it (at least, yet) 
![does not have](https://github.com/user-attachments/assets/414440a5-85b1-415f-96f4-148f9681ab84)
4. In addition to the p3: Recover funds tokens list shows regular balanes in this case: 
![no funds](https://github.com/user-attachments/assets/a4707919-2521-432e-919b-39f46d9470fa)
![when no, regular balances](https://github.com/user-attachments/assets/5ecfa759-ccd8-44e4-b2c1-7112efa8f65b)
5. Confirm modal: warnings also show a proxy address that an account does not have (at least, yet): 
![will be deployed](https://github.com/user-attachments/assets/d2a41104-bed2-441b-b160-d38c9df87a32)
In this case, I think, we need a bit different message (cc @MindyCoW )
![Screenshot_1](https://github.com/user-attachments/assets/d3b57834-3473-4010-a6c0-99f18772f9a5)
6. Could you please add a link to the 'recover funds' text? 
![link on tab](https://github.com/user-attachments/assets/bd8890ce-38ef-43f3-b598-d8996c88dda0)
7. In terms of UI, could you please copy the warning pattern from the one we have on Linit orders? 
![padd](https://github.com/user-attachments/assets/eb713a38-7902-45ee-887d-6598c49b13ed)
![image](https://github.com/user-attachments/assets/b3fbec09-9fe8-4778-be7d-31c0d883fd77)
8. When I press on the 'Connect to a wallet' button from the Proxy account page, it navigates me to the Swap,. Expected: stay on the Recover funds page. 
9. When I press on the 'Read more' from the Swap widget, the quote request runs and shows me the quote loading state first, then shows me Proxy account page. See the video: https://www.loom.com/share/0b6f0e7b416c49ceb52e6ed8aaf661c6
10. When I on the Proxy account page, with every mouse click the page is closed and I get to the Swap page
11. In terms of navigation, it would be great to navigate a user to the previous page they came from. E.g. if I come from Tokens page, pressing on the back button should return me there, not to the Swap widget. 
12. Still, I think that it would be nice to separate FAQ questions between tabs on the page . Was it discussed in the UX sync? Or it does not worth it? 
![image](https://github.com/user-attachments/assets/b978ddc1-e275-4892-ab3f-16291aeaec8d)

Thanks

shoom3301 commented:
>let's skip it

@elena-zh we can improve those points after, now we need to give Mindy some version for user testing.

elena-zh reviewed (approved):
Hey @shoom, thanks. 
I've opened these tasks to address them later: 
1. (new)  https://www.notion.so/cownation/Account-Proxy-page-does-not-cover-the-screen-in-a-mobile-view-21e8da5f04ca800abeadfa93b7b4f752
2. https://www.notion.so/cownation/Account-Proxy-navigation-issues-21e8da5f04ca8079a621f90905c0997c
3. https://www.notion.so/cownation/Account-Proxy-separate-FAQ-between-tabs-21e8da5f04ca80289b50e0a7ec5af938
4. https://www.notion.so/cownation/Account-Proxy-Add-link-to-the-warning-21e8da5f04ca805db541c68c2d7b16ca
